csrc/host_ir/jit.cpp

```diff
@@ -664,6 +690,42 @@ void compileFunctionDeclarations(
       kLinearOutFuncName,
       module);
 
+  // matmul function: at::Tensor* matmul(at::Tensor* a, at::Tensor* b)
+  auto* matmul_type = llvm::FunctionType::get(
```

Comment:
I've yet to understand how this change (and several other changes in this file) is related to the description of this PR. Did you intend to separate it?